apps/chat/src/components/Common/ApplicationWizard/ApplicationWizard.tsx

```diff
@@ -54,30 +63,44 @@ export const ApplicationWizard: React.FC<ApplicationWizardProps> = ({
     }
   }, [type]);
 
+  useEffect(() => {
+    if (submitted && loadingStatus === UploadStatus.LOADED) {
+      onClose(true);
+      setSubmitted(false);
+    } else if (submitted && loadingStatus === UploadStatus.FAILED) {
+      setSubmitted(false);
+    }
+  }, [loadingStatus, submitted, onClose]);
+
   return (
     <Modal
       portalId="theme-main"
       state={isOpen ? ModalState.OPENED : ModalState.CLOSED}
       onClose={handleClose}
       dataQa="application-dialog"
-      containerClassName="flex w-full flex-col pt-2 md:grow-0 xl:max-w-[720px] 2xl:max-w-[780px] !bg-layer-2"
+      containerClassName="flex w-full flex-col pt-2 md:grow-0 xl:max-w-[720px] 2xl:max-w-[780px] !bg-layer-2 relative"
       dismissProps={MOUSE_OUTSIDE_PRESS_EVENT}
       hideClose
     >
-      {isLoading ? (
+      {submitted && (
+        <div className="absolute left-0 top-0 z-10 flex size-full items-center justify-center bg-layer-2">
+          <Spinner size={48} dataQa="publication-items-spinner" />
+        </div>
+      )}
+      {isLoading && !submitted ? (
```

Comment:
`setSubmitted(true)` executed immediatelly
Should be like this for spinner
```suggestion
      {(isLoading || submitted) ? (
```